synse/websocket.go

```diff
@@ -0,0 +1,515 @@
+package synse
+
+// websocket.go implements a websocket client.
+
+import (
+	"crypto/tls"
+	"fmt"
+	"sync/atomic"
+
+	"github.com/gorilla/websocket"
+	"github.com/pkg/errors"
+	"github.com/vapor-ware/synse-client-go/synse/scheme"
+)
+
+// counter counts the number of request sent. It has the type uint64 which can
+// later be used by an atomic function, which makes it more concurrency-safe.
+// FIXME - is it necessary though?
+var counter uint64
```

Comment:
It is necessary for us to have some kind of atomic incremental counter so we can set the IDs correctly on the messages. This approach should work.

I will say that this variable should be moved out of the global scope and into the `websocketClient` scope so that it is tied to a single client's session. if its kept global and we initialize multiple sessions, they will all increment the same counter